### DISCUSSION TITLE
Why not use maplibre or leaflet instead of mapbox for Mapzy ?

### DISCUSSION THREAD

**Discussion by Pranav4399** [mapzy/mapzy#87] [Technical discussions]
Hello ! I'm a geospatial developer who primarily works on frontend stuff. Found this repo through HN and the use case seemed interesting. But while going through the docs, I found mapzy uses mapbox. Just wanted to know why did the creators went ahead with mapbox when libraries like leaflet/maplibre js exists ?

**Comment by shafy** [comment]
Hey @Pranav4399, this is a fair question. The simple (yet unsatisfactory) answer is that we wanted to get up and running quickly, and since we also use Mapbox as our mapping provider, going with their frontend SDK seemed the easiest. However, we've been looking into Leaflet and MapLibre, and we plan on switching over in the near future. We will for now keep Mapbox as the map server, but also have been looking at alternatives like self-hosting with MapTiler or a similar solution.

I've worked with Leaflet before, but never tried Maplibre. Do you have any opinions on Leaflet vs. Maplibre?

**Reply by Pranav4399** [reply]
Understood. From my experiences, I believe leaflet documentation is top notch but from my limited usage of mapLibreJs, it looks better compared to leaflet. If I remember correctly, maplibre was created as an open source alternative specifically for mapbox.

So, if the project is planning to switch, my personal choice would be to explore maplibre as I feel it has potential to grow.

Also, do you guys have a community in discord or anything else ?. I'd love to contribute to this project as well but it'll be easier for everyone to have a simpler discussion forum like discord or element.

**Reply by shafy** [reply]
We don't have a Discord or Element, we just use GitHub Discussions for these kind of conversations for now (I've created a new category "Technical discussion" and moved this discussion also over there). That's great to hear that you would want to contribute - anything specific in mind?

Would switching to MapLibre it in anyway change the UX or UI of Mapzy? For example, on the top right we have the location search with the Mapbox geocoder that's provided of out the box by Mapbox GL JS. I couldn't find anything comporable in MapLibre. Any thoughts on this?

![CleanShot 2022-09-05 at 18 04 13@2x](https://user-images.githubusercontent.com/1881676/188486786-5d0e115e-ccae-4a4c-a1a9-cccb68b0e66a.jpg)

**Reply by Pranav4399** [reply]
I'm yet to setup the code base and will look into the issues sections once that's done. Regarding geocoding in maplibre/leaflet, I need to look into them and I'm pretty sure, someone would've already thought about that and has a plugin built. 

The reason, why I'm interested in transitioning to a open source is that this project wouldn't be considered fully open source by someone trying to use it just because of the name mapbox being associated with it. I myself is a huge fan of what mapbox does to the community and what they're building but at the end of the day, they're still a proprietary company.

I'll update the same thread in case if I find a geocoding alternative.

Edit: Maplibre themselves has built the geocoder for the library already. Please look into this `https://github.com/maplibre/maplibre-gl-geocoder`

**Reply by shafy** [reply]
Sounds good!

> The reason, why I'm interested in transitioning to a open source is that this project wouldn't be considered fully open source by someone trying to use it just because of the name mapbox being associated with it

Yes, that's a valid point.

The MapLibre geocoder plugin looks good. We would need to combine it with a geocoding API, since the plugin itself doesn't provide the actual geocoding.
